src/NetAnalyzers/Microsoft.CodeAnalysis.NetAnalyzers.md

```diff
@@ -1620,6 +1620,30 @@ Number of parameters supplied in the logging message template do not match the n
 |CodeFix|False|
 ---
 
+## [CA2019](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2019): Do not pass mutable value types by value
```

Comment:
For both of these, what is the plan for handling and/or annotating value types which explicitly should be passed and returned by value?

For example, this is going to have tons of false positives in interop code. It's likewise not going to be taking SIMD (`Vector2`, `Vecto3`, or `Vector4`) or ABI (Unix can pass a struct in multiple registers sometimes) into account where passing by `ref` or `in` is generally inappropriate and may hurt perf or cause other regressions. This may likewise flag things like `Guid` incorrectly on older TFMs given that it wasn't updated to be `readonly` until newer releases (even though it was "functionally readonly" already).

Reply:
As per [this post](https://github.com/dotnet/runtime/issues/33772#issuecomment-913721146), this analyzer only flags value types that one of the following:

- On a built-in list of problematic types (the list currently contains `SpinLock` and `Utf8JsonReader`)
- Added as problematic mutable value types via an .editorconfig option,
- Types that it determines to be struct-enumerator types (must be a nested value type who's name ends with "Enumerator").